testing/scenariobuilder/validations.go

```diff
@@ -65,10 +87,21 @@ func validateAuthTokensCache(tokensCache string, url, expectedToken string) vali
 	}
 }
 
-func indexOfMatchingTodoScenario(scenarios []*TodoErrScenario, target string) int {
+func indexOfMatchingTodoScenario(scenarios []*TodoErrScenario, target string, format string) int {
 	for i := range scenarios {
-		if scenarios[i].String() == target {
-			return i
+		if format == "json" {
+			bytes, err := scenarios[i].ToJSON()
+			if err != nil {
+				panic(err)
+			}
+
+			if string(bytes) == target {
+				return i
+			}
```

Comment:
When you do the suggested refactoring of the functions, you can validate the json scenarios by simply marshalling all scenarios to JSON and comparing it against the program output.